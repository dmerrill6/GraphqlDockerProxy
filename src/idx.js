//@flow


global.idx = (obj, callBack) => {
  try {
    const res = callBack(obj)
    if (res === undefined){
      return null
    }
    return res
  } catch (e){
    return null
  }
}
