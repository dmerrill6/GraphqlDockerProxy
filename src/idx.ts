
declare module NodeJS  {
  interface Global {
    idx: (obj: any, callBack: (obj: any) => any) => any
    winston: any
  }
}
global.idx = (obj, callBack) => {
  try {
    const res = callBack(obj)
    if (res === undefined) {
      return null
    }
    return res
  } catch (e) {
    return null
  }
}
