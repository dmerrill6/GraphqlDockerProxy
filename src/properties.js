export const network = () => {
  return process.env.dockerNetwork || 'web'
}

export const token = () => {
  return idx(process, _ => _.env.gqlProxyToken) || ''
}

export const getVersion = () => {
  return idx(process, _ => _.env.VERSION)
}

export const getBuildNumber = () => {
  return idx(process, _ => _.env.BUILD_NUMBER)
}

/**
 * Available values: docker & kubernetes && kubernetesWatch
 */
export const runtime = () => {
  return idx(process, _ => _.env.qglProxyRuntime) || 'docker'
}

/**
 * true or false (false  = default)
 * If a backend is not reachable anymore the schema will be allready known
 * WIP Not produktionable
 */
export const knownOldSchemas = () => {
  return idx(process, _ => _.env.gqlProxyKnownOldSchemas) || false
}

export const kubernetesConfigurationKind = () => {
  // $kubernetesConfigurationKind
  /**
   * fromKubeconfig, getInCluster, getInClusterByUser
   */
  return idx(process, _ => _.env.kubernetesConfigurationKind) || 'fromKubeconfig'
}

export const k8sUser = () => {
  return idx(process, _ => _.env.gqlProxyK8sUser) || ''
}

export const k8sUserPassword = () => {
  return idx(process, _ => _.env.gqlProxyK8sUserPassword) || ''
}

export const getPollingMs = () => {
  return idx(process, _ => _.env.gqlProxyPollingMs) || 5000
}

export const adminUser = () => {
  return idx(process, _ => _.env.gqlProxyAdminUser) || ''
}

export const adminPassword = () => {
  return idx(process, _ => _.env.gqlProxyAdminPassword) || ''
}

export const showPlayground = () => {
  return idx(process, _ => _.env.gqlShowPlayground) || true
}


export const getBodyParserLimit = () => {
  return idx(process, _ => _.env.gqlBodyParserLimit) || '1mb'
}

export const printAllConfigs = () => {
  console.log('===================================')
  console.log('qglProxyRuntime:', runtime())
  console.log('gqlProxyPollingMs:', getPollingMs())
  console.log('gqlProxyAdminUser:', adminUser())
  console.log('gqlProxyKnownOldSchemas', knownOldSchemas())
  console.log('gqlShowPlayground', showPlayground())
  console.log('Version: ', getVersion())
  console.log('Buildnumber: ', getBuildNumber())
  if (runtime() === 'docker'){
    console.log('dockerNetwork:', network())
    console.log('gqlProxyToken:', token())
  } else if (runtime() === 'kubernetes'){
    console.log('kubernetesConfigurationKind:', kubernetesConfigurationKind())
    if (kubernetesConfigurationKind() === 'getInClusterByUser'){
      console.log('gqlProxyK8sUser:', k8sUser())
      console.log('gqlProxyK8sUserPassword:', '********')
    }
  }
  console.log('===================================')
}
