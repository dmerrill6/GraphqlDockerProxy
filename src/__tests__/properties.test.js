import * as properties from '../properties'
import '../idx'
global.console = {
  log: (...args) => {
    expect(args).toMatchSnapshot()
  },
}

describe('tests Properties', () => {
  const data = {
    dockerNetwork: 'mock',
    gqlProxyToken: '123',
    VERSION: '0.0.0',
    BUILD_NUMBER: '12345',
    qglProxyRuntime: 'mockRuntime',
    gqlProxyKnownOldSchemas: true,
    kubernetesConfigurationKind: 'getInCluster',
    gqlProxyK8sUser: 'mocktesting',
    gqlProxyK8sUserPassword: 'topSecret',
    gqlProxyPollingMs: 10124,
    gqlProxyAdminUser: 'mockAdmin',
    gqlProxyAdminPassword: 'topSecretAdmin',
  }
  beforeAll(() => {

  })
  it('snapshot all properties', () => {
    expect(properties).toMatchSnapshot()
  })

  for (const one in properties){
    it ('snapshot ' + one + ' default return', () => {
      expect(properties[one]()).toMatchSnapshot()
    })
    it ('snapshot ' + one + ' given envs', () => {
      process.env = Object.assign({}, process.env, data)
      expect(properties[one]()).toMatchSnapshot()
      process.env = {}
    })
  }
})
