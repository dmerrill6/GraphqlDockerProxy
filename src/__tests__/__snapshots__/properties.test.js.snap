// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Properties snapshot adminPassword default return 1`] = `""`;

exports[`tests Properties snapshot adminPassword given envs 1`] = `"topSecretAdmin"`;

exports[`tests Properties snapshot adminUser default return 1`] = `""`;

exports[`tests Properties snapshot adminUser given envs 1`] = `"mockAdmin"`;

exports[`tests Properties snapshot all properties 1`] = `
Object {
  "adminPassword": [Function],
  "adminUser": [Function],
  "getBuildNumber": [Function],
  "getPollingMs": [Function],
  "getVersion": [Function],
  "k8sUser": [Function],
  "k8sUserPassword": [Function],
  "knownOldSchemas": [Function],
  "kubernetesConfigurationKind": [Function],
  "network": [Function],
  "printAllConfigs": [Function],
  "runtime": [Function],
  "token": [Function],
}
`;

exports[`tests Properties snapshot getBuildNumber default return 1`] = `null`;

exports[`tests Properties snapshot getBuildNumber given envs 1`] = `"12345"`;

exports[`tests Properties snapshot getPollingMs default return 1`] = `5000`;

exports[`tests Properties snapshot getPollingMs given envs 1`] = `10124`;

exports[`tests Properties snapshot getVersion default return 1`] = `null`;

exports[`tests Properties snapshot getVersion given envs 1`] = `"0.0.0"`;

exports[`tests Properties snapshot k8sUser default return 1`] = `""`;

exports[`tests Properties snapshot k8sUser given envs 1`] = `"mocktesting"`;

exports[`tests Properties snapshot k8sUserPassword default return 1`] = `""`;

exports[`tests Properties snapshot k8sUserPassword given envs 1`] = `"topSecret"`;

exports[`tests Properties snapshot knownOldSchemas default return 1`] = `false`;

exports[`tests Properties snapshot knownOldSchemas given envs 1`] = `true`;

exports[`tests Properties snapshot kubernetesConfigurationKind default return 1`] = `"fromKubeconfig"`;

exports[`tests Properties snapshot kubernetesConfigurationKind given envs 1`] = `"getInCluster"`;

exports[`tests Properties snapshot network default return 1`] = `"web"`;

exports[`tests Properties snapshot network given envs 1`] = `"mock"`;

exports[`tests Properties snapshot printAllConfigs default return 1`] = `
Array [
  "===================================",
]
`;

exports[`tests Properties snapshot printAllConfigs default return 2`] = `
Array [
  "qglProxyRuntime:",
  "docker",
]
`;

exports[`tests Properties snapshot printAllConfigs default return 3`] = `
Array [
  "gqlProxyPollingMs:",
  5000,
]
`;

exports[`tests Properties snapshot printAllConfigs default return 4`] = `
Array [
  "gqlProxyAdminUser:",
  "",
]
`;

exports[`tests Properties snapshot printAllConfigs default return 5`] = `
Array [
  "gqlProxyKnownOldSchemas",
  false,
]
`;

exports[`tests Properties snapshot printAllConfigs default return 6`] = `
Array [
  "Version: ",
  null,
]
`;

exports[`tests Properties snapshot printAllConfigs default return 7`] = `
Array [
  "Buildnumber: ",
  null,
]
`;

exports[`tests Properties snapshot printAllConfigs default return 8`] = `
Array [
  "dockerNetwork:",
  "web",
]
`;

exports[`tests Properties snapshot printAllConfigs default return 9`] = `
Array [
  "gqlProxyToken:",
  "",
]
`;

exports[`tests Properties snapshot printAllConfigs default return 10`] = `
Array [
  "===================================",
]
`;

exports[`tests Properties snapshot printAllConfigs default return 11`] = `undefined`;

exports[`tests Properties snapshot printAllConfigs given envs 1`] = `
Array [
  "===================================",
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 2`] = `
Array [
  "qglProxyRuntime:",
  "mockRuntime",
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 3`] = `
Array [
  "gqlProxyPollingMs:",
  10124,
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 4`] = `
Array [
  "gqlProxyAdminUser:",
  "mockAdmin",
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 5`] = `
Array [
  "gqlProxyKnownOldSchemas",
  true,
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 6`] = `
Array [
  "Version: ",
  "0.0.0",
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 7`] = `
Array [
  "Buildnumber: ",
  "12345",
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 8`] = `
Array [
  "===================================",
]
`;

exports[`tests Properties snapshot printAllConfigs given envs 9`] = `undefined`;

exports[`tests Properties snapshot runtime default return 1`] = `"docker"`;

exports[`tests Properties snapshot runtime given envs 1`] = `"mockRuntime"`;

exports[`tests Properties snapshot token default return 1`] = `""`;

exports[`tests Properties snapshot token given envs 1`] = `"123"`;
